runner/run/processors/port_processor.py

```diff
@@ -121,6 +125,37 @@ def _convert_to_path(val):
         file_obj['path'] = path
         return file_obj
 
+    @staticmethod
+    def _covert_to_cwl_format(val):
+        file_obj = copy.deepcopy(val)
+        location = file_obj.pop('location',None)
+        if location:
+            try:
+                file_db_object = FileProcessor.get_file_obj(location)
+            except FileHelperException as e:
+                raise PortProcessorException('File %s not found' % location)
+            path = file_db_object.path
+            if path:
+                path_obj = Path(path)
+                if not file_obj.get('checksum'):
```

Comment:
I don't think we should check file_obj for checksum because, when we register output files we are always saving the checksum [here](https://github.com/mskcc/beagle/blob/master/runner/run/processors/port_processor.py#L159). [This](https://github.com/mskcc/beagle/pull/462/files#diff-7391770a3e6b593d59c60681987308cfR141-R143) is good enough. Same thing for input files, we are using values from File object in DB to create a job.